ecc_enclave/enclave/crypto.cpp

```diff
@@ -111,37 +109,46 @@ std::string extract_encoded_public_key(const ByteArray cert_pem)
     EVP_PKEY* pubkey = X509_get_pubkey(cert);
     EC_KEY* eckey = EVP_PKEY_get1_EC_KEY(pubkey);
 
-    BIO_ptr pkBio(BIO_new(BIO_s_mem()), BIO_free_all);
-    PEM_write_bio_EC_PUBKEY(pkBio.get(), eckey);
-    int keylen = BIO_pending(pkBio.get());
+    BIO* pkBio = BIO_new(BIO_s_mem());
```

Comment:
The return value is not checked. Should doublecheck whether it fails properly later in case of error.

Reply:
added